### PR TITLE
Rebase of #3706

### PR DESCRIPTION
**Pull request by smowton**
https://github.com/github/codeql/pull/3706/

### CODE REVIEW

**Comment by aschackmull** [comment]
```
[ERROR] ql/java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql: expected result pattern(s) are not present for query kind "path-problem"
```

File: java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ * @tags security
+ *       external/cwe/cwe-749
+ *       external/cwe/cwe-079
+ */
+
+import java
+import semmle.code.java.frameworks.android.Intent
+import semmle.code.java.frameworks.android.WebView
+import semmle.code.java.dataflow.FlowSources
+
```
Comment (aschackmull) [inline java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql:14]: Looks like this is missing `import DataFlow::PathGraph`.

File: java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql [hunk 2]
```diff
@@ -0,0 +1,146 @@
+/**
+ * @name Unsafe resource fetching in Android webview
+ * @description JavaScript rendered inside WebViews can access any protected application file and web resource from any origin
```
Comment (aibaars) [inline java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql:3]: Please add an `@id` property to the query. For example:
```suggestion
 * @id java/android/unsafe-resource-fetching
 * @description JavaScript rendered inside WebViews can access any protected application file and web resource from any origin
```

**Comment by smowton** [comment]
Thanks @aschackmull @aibaars, both of those are applied.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**